pkg/cmd/run.go

```diff
@@ -462,19 +462,6 @@ func fetchSecrets(localConfig models.ScopedOptions, enableCache bool, fallbackOp
 			}
 		}
 
-		// TODO remove this when releasing CLI v4 (DPLR-435)
-		if fallbackOpts.legacyPath != "" && localConfig.EnclaveProject.Value != "" && localConfig.EnclaveConfig.Value != "" {
-			utils.LogDebug(fmt.Sprintf("Writing to legacy fallback file %s", fallbackOpts.legacyPath))
-			if err := utils.WriteFile(fallbackOpts.legacyPath, []byte(encryptedResponse), utils.RestrictedFilePerms()); err != nil {
-				utils.Log("Unable to write to legacy fallback file")
-				if fallbackOpts.exitOnWriteFailure {
-					utils.HandleError(err, "", strings.Join(writeFailureMessage(), "\n"))
-				} else {
-					utils.LogDebugError(err)
-				}
-			}
-		}
```

Comment:
This is technically a breaking change in that `doppler run` performed with the latest client won't generate a fallback file that CLI v3.14.0 (and below) can understand. Given that [v3.14.0](https://github.com/DopplerHQ/cli/releases/tag/3.14.0) shipped over 2 years ago, I think this risk is acceptable.

Reply:
I think this is an acceptable level of risk so going to remain on v3. Major version upgrades shouldn't be a big deal, though they've traditionally been annoying as a lot of folks pin to a specific major version in their dockerfiles.

Though given that we've moved away from recommending the docker image as the default method (we recommend install.sh or using your OS's package manager), this probably isn't as big a deal as it used to be.